tests/unit/mathematics/linalg/operations/Viennacl_operations_unittest.cc

```diff
@@ -49,4 +49,23 @@ TEST(LinalgBackendViennaCL, dot)
 	EXPECT_NEAR(result, 5, 1E-15);
 }
 
+TEST(LinalgBackendViennaCL, set_const)
+{
+	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
+
+	const index_t size = 5, value = 2;
+	SGVector<int32_t> a(size), b(size);
+	SGVector<int32_t> b_gpu, b_cpu;
+
+	for (index_t i = 0; i < size; ++i)
+		a[i] = value;
+
+	b_gpu = to_gpu(b);
+	set_const(b_gpu, value);
+	b_cpu = from_gpu(b_gpu);
+
+	for (index_t i = 0; i < size; ++i)
+		EXPECT_NEAR(a[i], b_cpu[i], 1E-15);
+}
```

Comment:
do you get "reading uninitialized values" error on valgrind?
however, for this unit-test's purpose, you can set the value const to 0 for the cpu vector, transfer to gpu, set the value to something else, transfer back and test. that will get rid of this error.